catalog/ui/catalog-ui-search/src/main/webapp/js/CQLUtils.js

```diff
@@ -98,7 +98,9 @@ define([
             var cqlPolygon = model.map(function (point) {
                 return point[0] + ' ' + point[1];
             });
-            cqlPolygon.push(cqlPolygon[0]);
+            if (cqlPolygon[0] !== cqlPolygon[cqlPolygon.length - 1]) {
+                cqlPolygon.push(cqlPolygon[0]);
```

Comment:
Does this remove the first point if it doesn't match the last point? I feel like the extra points were being generated at the end and we should be trimming duplicates off the end? Correct me if I'm wrong 😄